workflow-core/src/main/java/com/squareup/workflow1/StatefulWorkflow.kt

```diff
@@ -166,6 +150,164 @@ abstract class StatefulWorkflow<
+    update: WorkflowAction<PropsT, StateT, OutputT>.Updater.(E1, E2, E3, E4, E5) -> Unit
+  ): (E1, E2, E3, E4, E5) -> Unit {
+    return { e1, e2, e3, e4, e5 ->
+      val action = object : NamedAction<PropsT, StateT, OutputT>(name) {
+        override fun Updater.apply() = update(e1, e2, e3, e4, e5)
+      }
+
+      actionSink.send(action)
+    }
+  }
+
+  fun <E1, E2, E3, E4, E5, E6> RenderContext.eventHandler(
+    name: String = "eventHandler",
+    update: WorkflowAction<PropsT, StateT, OutputT>.Updater.(E1, E2, E3, E4, E5, E6) -> Unit
+  ): (E1, E2, E3, E4, E5, E6) -> Unit {
+    return { e1, e2, e3, e4, e5, e6 ->
+      val action = object : NamedAction<PropsT, StateT, OutputT>(name) {
+        override fun Updater.apply() = update(e1, e2, e3, e4, e5, e6)
+      }
+
+      actionSink.send(action)
+    }
+  }
+
+  fun <E1, E2, E3, E4, E5, E6, E7> RenderContext.eventHandler(
+    name: String = "eventHandler",
+    update: WorkflowAction<PropsT, StateT, OutputT>.Updater.(E1, E2, E3, E4, E5, E6, E7) -> Unit
+  ): (E1, E2, E3, E4, E5, E6, E7) -> Unit {
+    return { e1, e2, e3, e4, e5, e6, e7 ->
+      val action = object : NamedAction<PropsT, StateT, OutputT>(name) {
+        override fun Updater.apply() = update(e1, e2, e3, e4, e5, e6, e7)
+      }
+
+      actionSink.send(action)
+    }
+  }
+
+  fun <E1, E2, E3, E4, E5, E6, E7, E8> RenderContext.eventHandler(
+    name: String = "eventHandler",
+    update: WorkflowAction<PropsT, StateT, OutputT>.Updater.(E1, E2, E3, E4, E5, E6, E7, E8) -> Unit
+  ): (E1, E2, E3, E4, E5, E6, E7, E8) -> Unit {
+    return { e1, e2, e3, e4, e5, e6, e7, e8 ->
+      val action = object : NamedAction<PropsT, StateT, OutputT>(name) {
+        override fun Updater.apply() = update(e1, e2, e3, e4, e5, e6, e7, e8)
+      }
+
+      actionSink.send(action)
+    }
+  }
+
+  fun <E1, E2, E3, E4, E5, E6, E7, E8, E9> RenderContext.eventHandler(
+    name: String = "eventHandler",
+    update: WorkflowAction<PropsT, StateT, OutputT>
+    .Updater.(E1, E2, E3, E4, E5, E6, E7, E8, E9) -> Unit
+  ): (E1, E2, E3, E4, E5, E6, E7, E8, E9) -> Unit {
+    return { e1, e2, e3, e4, e5, e6, e7, e8, e9 ->
+      val action = object : NamedAction<PropsT, StateT, OutputT>(name) {
+        override fun Updater.apply() = update(e1, e2, e3, e4, e5, e6, e7, e8, e9)
+      }
+
+      actionSink.send(action)
+    }
+  }
+
+  fun <E1, E2, E3, E4, E5, E6, E7, E8, E9, E10> RenderContext.eventHandler(
+    name: String = "eventHandler",
+    update: WorkflowAction<PropsT, StateT, OutputT>
+    .Updater.(E1, E2, E3, E4, E5, E6, E7, E8, E9, E10) -> Unit
+  ): (E1, E2, E3, E4, E5, E6, E7, E8, E9, E10) -> Unit {
+    return { e1, e2, e3, e4, e5, e6, e7, e8, e9, e10 ->
+      val action = object : NamedAction<PropsT, StateT, OutputT>(name) {
+        override fun Updater.apply() = update(e1, e2, e3, e4, e5, e6, e7, e8, e9, e10)
+      }
+
+      actionSink.send(action)
+    }
+  }
+
+  private abstract class NamedAction<PropsT, StateT, Output>(
```

Comment:
There's a dup of this in `StatelessWorkflow`. Should be able to share something.